t/lib/OpenQA/Test/Utils.pm

```diff
@@ -454,7 +454,9 @@ sub unstable_worker {
     # the help of the Doctor would be really appreciated here.
     my ($apikey, $apisecret, $host, $instance, $ticks, $sleep) = @_;
     note("Starting unstable worker. Instance: $instance for host $host");
-    $ticks = 1 unless defined $ticks;
+    # Block the event loop for the specified number of events or indefinitely if -1 is specified
+    $ticks = $ticks == -1 ? ~0 >> 1 : ($ticks // 1);
```

Comment:
I don't understand this. Isn't `-1` always true?

Reply:
this should be equivalent to:

```perl
if ($ticks == -1) $ticks = ~0 >> 1;
else { $ticks = $ticks // 1 };
```

Reply:
oh right. Stupid me missed the `==`.

Reply:
I don't find `~0 >> 1` particularly readable. How about just using a boring old constant?

Reply:
Does the method accept `'inf'` as a value for the ticks? :)

Reply:
I'm assuming the worker here is unstable because it only runs a few ticks and will therefore just vanish at some point while processing a job. It's a bit sketchy, but if the number is low enough should be reliable enough because I/O events will be spaced out enough. If it were relying more on timers this would be super inconsistent and unstable.

A much better strategy would have been to always use `Mojo::IOLoop->start` and then hook into the worker instance to call `Mojo::IOLoop->stop` at a very specific point to simulate an unstable worker more consistently.

Reply:
Really not a fan of relying on randomness for tests.

Reply:
External factors like the size of the kernel send buffer will influence what this test does, because ticks are mostly one I/O event per socket. So the more packets there are to read the more ticks you waste with it... and so on. Not a great strategy. I'd rather monkey patch a `Mojo::IOLoop->stop` into the worker.